common/discovery.go

```diff
@@ -51,17 +51,22 @@ func DiscoverNodesForService(consul *api.Client, serviceName string) ([]Node, er
 
 	var nodeList []Node
 	for _, svc := range services {
-		var addr = svc.Node.Address
+		// Node Address is always the IP of the host running consul
+		// whereas Service address is the actual IP of the service.
+		// With Cilium, we need the Service address (which is the Pod address)
+		var addr = svc.Service.Address
 
-		node_name := svc.Node.Node
+		node_name := svc.Service.Meta["k8s_pod"]
+		host_name := svc.Node.Node
 		if fqdn, ok := svc.Node.Meta["fqdn"]; ok {
-			node_name = fqdn
+			host_name = fqdn
```

Comment:
why do you want to use fqdn meta ? when I look it doesn't correspond to any known reference on our side (ex: 10-184-106-4.container.crto.in).

Do we really need to have host name ? is node ip not sufficient ?